autoscale/src/main/java/com/sequenceiq/periscope/repository/ScalingActivityRepository.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.periscope.repository;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+import javax.transaction.Transactional;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+import com.sequenceiq.cloudbreak.workspace.repository.EntityType;
+import com.sequenceiq.periscope.api.model.ActivityStatus;
+import com.sequenceiq.periscope.domain.ScalingActivity;
+
+@EntityType(entityClass = ScalingActivity.class)
+@Transactional(Transactional.TxType.REQUIRED)
+public interface ScalingActivityRepository extends CrudRepository<ScalingActivity, Long> {
+
+    @Query("SELECT st FROM ScalingActivity st WHERE st.activityCrn = :activityCrn")
+    Optional<ScalingActivity> findByActivityCrn(@Param("activityCrn") String activityCrn);
+
+    @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId")
+    List<ScalingActivity> findAllByCluster(@Param("clusterId") Long clusterId);
+
```

Comment:
There's a good chance that we're going to have an edge case error here. There's 3 queries related to start_time.
1. >=startTime
2. <=startTime
3. >=startTime AND <startTime

2 and 3 are not consistent. Am guessing 2 will need to be changed to <startTime instead of <=startTime . To be seen in the actual usage.

Reply:
Updated this as suggested